docs/default.nix

```diff
@@ -32,74 +32,447 @@ let
 
   # TODO: Include Nix Darwin options
 
-  makeOptionsDoc =
-    { configuration, pathFilter }:
-    (pkgs.nixosOptionsDoc {
-      inherit (configuration) options;
-      transformOptions =
+  platforms = {
+    home_manager = {
+      name = "Home Manager";
+      configuration = homeManagerConfiguration;
+    };
+    nixos = {
+      name = "NixOS";
+      configuration = nixosConfiguration;
+    };
+  };
+
+  # We construct an index of all Stylix options, using the format:
+  #
+  # {
+  #   "src/options/modules/«module».md" = {
+  #     referenceSection = "Modules";
+  #     readme = "modules/«module»/README.md";
+  #     defaultReadme = "note about the path above not existing";
+  #     optionsByPlatform = {
+  #       home_manager = [ ... ];
+  #       nixos = [ ... ];
+  #     };
+  #   };
+  #
+  #   "src/options/platforms/«platform».md" = {
+  #     referenceSection = "Platforms";
+  #     readme = "docs/src/options/platforms/«platform».md";
+  #     defaultReadme = "note about the path above not existing";
+  #     optionsByPlatform.«platform» = [ ... ];
+  #   };
+  # }
+  #
+  # Options are inserted one at a time into the appropriate page, creating
+  # new page entries if they don't exist.
+
+  insert =
+    {
+      index,
+      page,
+      emptyPage,
+      platform,
+      option,
+    }:
+    index
+    // {
```

Comment:
It is probably overkill to use `lib.attrsets.unionOfDisjoint` here, right?

Reply:
We are deliberately overwriting existing pages to insert new items, so this would not be possible.